services/s3/src/main/java/software/amazon/awssdk/services/s3/LegacyMd5Plugin.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3;
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.core.SdkPlugin;
+import software.amazon.awssdk.core.SdkServiceClientConfiguration;
+import software.amazon.awssdk.core.checksums.RequestChecksumCalculation;
+import software.amazon.awssdk.core.checksums.ResponseChecksumValidation;
+import software.amazon.awssdk.services.s3.internal.handlers.LegacyMd5ExecutionInterceptor;
+/**
+ * Plugin that enables legacy MD5 checksum behavior for S3 operations.
+ *
+ * <p>This plugin configures the S3 client to:
+ * <ul>
+ *   <li>Set request checksum calculation to WHEN_REQUIRED mode, which calculates default checksums only when
+ *       required by the S3 service for specific operations</li>
+ *   <li>Set response checksum validation to WHEN_REQUIRED mode, which validates checksums only when
+ *       the S3 service provides them in responses</li>
+ *   <li>Add an interceptor that maintains backward compatibility with older SDK versions
+ *       that automatically calculated MD5 checksums for certain operations</li>
+ * </ul>
+ *
+ * <p>Use this plugin only when you need to maintain compatibility with applications that depend on the
+ * legacy MD5 checksum behavior, particularly for operations that previously calculated MD5 checksums
+ * automatically.
+ * <p><b>Example usage:</b>
+ * <p>
+ * {@snippet :
+ * // For synchronous S3 client
+ *  S3Client s3Client = S3Client.builder()
+ *                             .addPlugin(LegacyMd5Plugin.create())
+ *                             .build();
+ *
+ *  // For asynchronous S3 client
+ *  S3AsyncClient asyncClient = S3AsyncClient.builder()
+ *                                          .addPlugin(create())
```

Comment:
`LegacyMd5Plugin.create()`. Can we configure the request checksum calculation and response checksum validation to when_required in the code snippet?

Reply:
Done